src/components/Filter/types.ts

```diff
@@ -29,9 +29,17 @@ export type Props = {
   selectedItem?: FilterOption;
   /** Defines if the button is in disabled state */
   disabled?: boolean;
-};
-
-export type TestProps = {
+  /** if the options are searchable */
+  isSearchable?: boolean;
+  /** if the component is used asynchronously */
+  isAsync?: boolean;
+  /** the function to fetch new options */
+  onAsyncSearch?: (term: string) => void;
```

Comment:
Maybe symblify the props by having a `onSearch` function and a `isAsync` flag.  If on `onSearch`  exists the `isSearchable` would be true so it's overkill to have both.

Reply:
Thank you for the suggestion. I think is `isSearchable` is required but for sure we can have avoid `isAsync` and keep only `onAsyncSearch`. I will give a try.

When just `isSearchable` is true then the component will filter the options inside so no `onSearch` is needed. But when `onAsyncSearch` is passed then the options are not filtered, the component just triggers `onAsyncSearch`.

Reply:
@Christos-Zacharopoulos ok I changed it. please have a look when possible! thanks!

Reply:
![image](https://user-images.githubusercontent.com/28539607/123074569-19b0ef00-d420-11eb-916b-014cc00d3304.png)